osbs/build/build_requestv2.py

```diff
@@ -0,0 +1,292 @@
+                .format(ISOLATED_RELEASE_FORMAT.pattern))
+
+        self.set_label('isolated', 'true')
+        self.set_label('isolated-release', self.user_params.release.value)
+
+    def adjust_for_repo_info(self):
+        if not self._repo_info:
+            logger.warning('repo info not set')
+            return
+
+        if not self._repo_info.configuration.is_autorebuild_enabled():
+            logger.info('autorebuild is disabled in repo configuration, removing triggers')
+            self.template['spec'].pop('triggers', None)
+
+        else:
+            labels = Labels(self._repo_info.dockerfile_parser.labels)
+            try:
+                labels.get_name_and_value(Labels.LABEL_TYPE_RELEASE)
+            except KeyError:
+                # As expected, release label not set in Dockerfile
+                pass
+            else:
+                raise RuntimeError('when autorebuild is enabled in repo configuration, '
+                                   '"release" label must not be set in Dockerfile')
+
+    def render_name(self):
+        """Sets the Build/BuildConfig object name"""
+        name = self.user_params.name.value
+
+        if self.scratch or self.isolated:
+            name = self.user_params.image_tag.value
+            platform = self.user_params.platform.value
+            # Platform name may contain characters not allowed by OpenShift.
+            if platform:
+                platform_suffix = '-{0}'.format(platform)
+                if name.endswith(platform_suffix):
+                    name = name[:-len(platform_suffix)]
+
+            _, salt, timestamp = name.rsplit('-', 2)
+
+            if self.scratch:
+                name = 'scratch-{0}-{1}'.format(salt, timestamp)
+            elif self.isolated:
+                name = 'isolated-{0}-{1}'.format(salt, timestamp)
+
+        # !IMPORTANT! can't be too long: https://github.com/openshift/origin/issues/733
+        self.template['metadata']['name'] = name
+
+    def render_node_selectors(self):
+        # for worker builds set nodeselectors
+        if self.user_params.platforms.value is None:
+            # auto or explicit build selector
+            if self.is_auto:
+                self.template['spec']['nodeSelector'] = self.auto_build_node_selector
+            # scratch build nodeselector
+            elif self.scratch:
+                self.template['spec']['nodeSelector'] = self.scratch_build_node_selector
+            # isolated build nodeselector
+            elif self.isolated:
+                self.template['spec']['nodeSelector'] = self.isolated_build_node_selector
+            # explicit build nodeselector
+            else:
+                self.template['spec']['nodeSelector'] = self.explicit_build_node_selector
+
+            # platform nodeselector
+            if self.platform_node_selector:
+                self.template['spec']['nodeSelector'].update(self.platform_node_selector)
+
+    def render(self, api=None, validate=True):
```

Comment:
Should `api` be a required parameter?

Reply:
Technically yes, but making it a required parameter messes up the mock'd lambdas in test_api and I would rather not unsort that mess.

Reply:
Maybe we need to raise OsbsValidationError first of all if it is None, in that case.

Reply:
Fixed.

Reply:
in last part of reactor config map story, I will pass self.os, to set_params, and it will be available in BuildRequest as self.os

Reply:
so won't need to pass it to render() but you can just access it

Reply:
okay. I'll keep the scaffolding in for now and fix it on the merge.